browser/ui/webui/brave_rewards_page_ui.cc

```diff
@@ -538,6 +545,74 @@ void RewardsDOMHandler::Init() {
   rewards_service_->StartProcess(base::DoNothing());
 
   ads_service_ = brave_ads::AdsServiceFactory::GetForProfile(profile);
+
+  // Configure a pref change registrar to update brave://rewards when settings
+  // are changed via brave://settings
+  InitPrefChangeRegistrar();
+}
+
+void RewardsDOMHandler::InitPrefChangeRegistrar() {
+  Profile* profile = Profile::FromWebUI(web_ui());
+  pref_change_registrar_.Init(profile->GetPrefs());
+
+  pref_change_registrar_.Add(
+      ads::prefs::kEnabled,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      ads::prefs::kAdsPerHour,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      ads::prefs::kAdsSubdivisionTargetingCode,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kAutoContributeEnabled,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kAutoContributeAmount,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kMinVisitTime,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kMinVisits,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kAllowNonVerified,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kAllowVideoContribution,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kInlineTipTwitterEnabled,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kInlineTipRedditEnabled,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kInlineTipGithubEnabled,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+}
+
+void RewardsDOMHandler::OnPreferenceChanged(const std::string& key) {
+  AllowJavascript();
```

Comment:
Without changing anything else, we should probably no-op if we don't currently "allow" JS. Separately, I *think* the desired pattern is to only have observers registered from the `JavascriptAllowed` method.

Reply:
Ah yes, fixed.